yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/command/Command.java

```diff
@@ -127,26 +157,50 @@ private static Command getSetSchemaStateCommand(int lineNumber, @Nonnull final Y
             @Override
             public void executeInternal(@Nonnull RelationalConnection connection) throws SQLException, RelationalException {
                 logger.debug("⏳ Setting schema state '{}'", value);
-                StoreCatalog backingCatalog = ((EmbeddedRelationalConnection) connection).getBackingCatalog();
                 SchemaInstanceOuterClass.SchemaInstance schemaInstance = CommandUtil.fromJson(value);
                 RecordLayerConfig rlConfig = new RecordLayerConfig.RecordLayerConfigBuilder()
                         .setIndexStateMap(CommandUtil.fromIndexStateProto(schemaInstance.getIndexStatesMap()))
                         .setFormatVersion(schemaInstance.getStoreInfo().getFormatVersion())
                         .setUserVersionChecker((oldUserVersion, oldMetaDataVersion, metaData) -> CompletableFuture.completedFuture(schemaInstance.getStoreInfo().getUserVersion()))
                         .build();
-                RecordLayerMetadataOperationsFactory metadataOperationsFactory = new RecordLayerMetadataOperationsFactory.Builder()
-                        .setBaseKeySpace(RelationalKeyspaceProvider.instance().getKeySpace())
-                        .setRlConfig(rlConfig)
-                        .setStoreCatalog(backingCatalog)
-                        .build();
-
-                final var embeddedConnection = (EmbeddedRelationalConnection) connection;
-                embeddedConnection.setAutoCommit(false);
-                embeddedConnection.createNewTransaction();
-                final var transaction = embeddedConnection.getTransaction();
-                metadataOperationsFactory.getSetStoreStateConstantAction(URI.create(schemaInstance.getDatabaseId()), schemaInstance.getName()).execute(transaction);
-                embeddedConnection.commit();
-                embeddedConnection.setAutoCommit(true);
+                StoreCatalog backingCatalog;
+                if (connection instanceof EmbeddedRelationalConnection) {
+                    backingCatalog = ((EmbeddedRelationalConnection) connection).getBackingCatalog();
```

Comment:
I think if you extract two methods that each take a lambda, this could be something like:
```
var applyState = factory -> factory..getSetStoreStateConstantAction(URI.create(schemaInstance.getDatabaseId()), schemaInstance.getName()).execute(transaction);
if (connection instanceof EmbeddedRelationalConnection) {
    applyMetadataOperationEmbedded((EmbeddedRelationalConnection) connection, applyState);
} else {
    applyMetadataOperationDirectly(connection, applyState);
}
```

And then, I think, `getLoadSchemaTemplateCommand` could be similar, but with a different function.

Reply:
great idea, done.